### PR TITLE
Support Korean Hangul Syllables on han3_ji7_tsoo1_kian3

### PR DESCRIPTION
**Pull request by studioego**
That project's font files not supported Korean Hangul Syllable Characters, for example 가(U+AC00)-힣(D7A3).

I add two Korean Hangul Fonts, "Nanum Gothic" and "Nanum Myeongjo" on Project.
prefix "Nanum" is brand name.
"Gothic" is "고딕체/고딕體", reads "godikche" in Korean. I think The Chinese font 楷體 is equivalent to the Korean font Gothic 고딕체.
"Myeongjo" is "명조체/明朝體", reads " myeongjoche" in Korean. I think The Chinese font 宋體 is equivalent to the Korean font Myeongjo 명조체/明朝體.
Nanum-series Korean Font Official Link1: http://hangeul.naver.com/download.nhn
Nanum-series Korean Font Official Link2: http://hangeul.naver.com/2017/nanum

BTW, These Font's License is SIL Open Font License.
License Information Link: http://help.naver.com/ops/step2/faq.nhn?faqId=15879

Also, These fonts are default font on Ubuntu Korean Language distribution.

example) ⿴囗도
Ref: https://twitter.com/ursaqita/status/897102713007120384
⿴囗도
 "⿴囗도". This word origin from Chinese Character "圖".
"圖" reads "do" in Korean, write "도" in Korean Hangul.
I think that "⿴囗도"'s notation seems to be denoted to emphasizes Korean Sound "do"(도).

### CODE REVIEW

**Comment by hfhchan** [comment]
If Gothic is sans-serif, then the Chinese name is 黑體

**Comment by Artoria2e5** [comment]
Yep, the "proper" Sans-serif/黑 for Korean should look like https://typekit.com/fonts/source-han-sans-korean.

I am actually curious about Hangul Ming (Serif) / Kai (regular script) though: in most Korean serif fonts, Hangul glyphs come with some pretty script-like stroke variations, making them look somehow like Kai. (example: https://typekit.com/fonts/source-han-serif-korean) Is there a way to reliably tell a 楷書 from a 明朝體 only given the Hangul part? Or are they just supposed to be the same?

**Comment by studioego** [comment]
I think I misunderstood Korean fonts. 

> @hfhchan If Gothic is sans-serif, then the Chinese name is 黑體

Yes.  Korean font 고딕체[Gothic, Godik-che] is sans-serif, then the Chinese name is 黑體

> @Artoria2e5   Is there a way to reliably tell a 楷書 from a 明朝體 only given the Hangul part?

 No. Another is 궁서체[Gungseo-Che or gungsuh-che,宮書體].
 The 궁서체[Gungseo-Che or gungsuh-che,宮書體] is the abbreviation of the 궁중 서체[宮中書體/gungjung-seoche].
 It looks like  Korean 楷體 or 해서체[HaeSeo-Che,楷書體]. 
 The Font style popular among girls in the Korean palace since 500 years ago.

The  '해서체/楷書體Haeseo-che ' font is similar to the '궁서체[Gungseo-Che or gungsuh-che,宮書體]' font. 
In Korea, Korean mixed script[국한문혼용문(國漢文混用文), a form of writing that uses both Hangul(Hangeul, 한글) and Hanja/한자/漢字, Ideograph] mostly use both 해서체[HaeSeo-Che,楷書體] font on Hanja(漢字) and 궁서체[Gungseo-Che or gungsuh-che,宮書體] font on Hangul(한글).

This is 궁서체[Gungseo-Che or gungsuh-che,宮書體]'s description on Chinese Wikipedia.  https://zh.wikipedia.org/wiki/宮書體

It's Chinese font examples on LibreOffice. 
https://www.facebook.com/groups/libreozht.users/permalink/1634397096583409/

I'll check All Open Source font in Korean. @hfhchan @Artoria2e5 

After doing research on Korean Open Source fonts, I'll fix the code & font files.

Thank you for your advice.

This is Windows Korean Font's examples.
바탕체[Batang-Che,바탕體] -  Serif, [In the mid-1990s, Korean Ministry of Culture(문화부/文化部, Munhwa-bu) specified this term, replacing Myeongjo(명조체/明朝體,myeongjo-che]
돋음체[Dotum-Che,돋음體] - Sans-serif, It's Gothic's Korean name. [In the mid-1990s, Korean Ministry of Culture(문화부/文化部, Munhwa-bu) specified this term, replacing Gothic(고딕체/고딕體,godik-che]
굴림체[Gulim-Che,굴림體] - Sans-serif 
궁서체[Gungseo-Che or gungsuh-che,宮書體) - Maybe It looks like  Korean 楷體 or 해서체[HaeSeo-Che,楷書體].

[MS Unicode Fonts 서체（書體）Windows XP/Vista
](http://www.wildboar.net/multilingual/asian/korean/language/fonts/unicode/microsoft/microsoft.html
)

**Comment by Artoria2e5** [comment]
Ah, just take notice that the tests are broken because the Java version is now too low for gradle to downlad things. I think the fonts you are looking at are OK, as they are SIL OFL.

We are trying to figure out a way to not store the fonts in git, but that's not done yet, so feel free to just put the fonts alongside the source in this pull request.

File: src/main/java/cc/char_indexingtool/FontCorrespondTable.java [hunk 1]
```diff
@@ -22,6 +22,7 @@
 			"/font/CNS11643-101_11_28/TW-Kai-98_1.ttf",
 			"/font/CNS11643-101_11_28/TW-Kai-Ext-B-98_1.ttf",
 			"/font/CNS11643-101_11_28/TW-Kai-Plus-98_1.ttf",
+			"/font/NanumFont/NanumGothic.ttf",
```
Comment (Artoria2e5) [inline src/main/java/cc/char_indexingtool/FontCorrespondTable.java]: We should use NanumMyeongjo for both for now. NanumMyeongjo's Hangul looks more "Kai" than the Gothic.

Comment (Artoria2e5) [reply src/main/java/cc/char_indexingtool/FontCorrespondTable.java]: The Gothic part will be used after we figure out 黑體支援.

**Comment by studioego** [comment]
changed java file

**Comment by sih4sing5hong5** [comment]
@studioego Is the PR finished or WIP ?
If completed, we will merge this.

**Comment by studioego** [comment]
Yes, Completed it.  Please, merge it

**Pull request merged**